### PR TITLE
Add support for tie_word_embeddings when loading weights + support for SmolLM

### PR DESCRIPTION
**Pull request by TianyiQ**
## Motivation

When trying to add support for SmolLM, I noticed that its config dict has `"tie_word_embeddings": true` which isn't supported by SGLang, and as a result, `lm_head.weight` is not loaded. This problem likely exists for some other models too (not just SmolLM).

## Modifications

Within `LlamaForCausalLM`, at the end of the weight loading stage, I added the following code:

```python3
if hasattr(self.config, "tie_word_embeddings") and self.config.tie_word_embeddings:
    # Tie output embedding layer to input embedding layer, to solve issues where lm_head.weight is missing
    param = self.lm_head.weight
    weight_loader = getattr(param, "weight_loader", default_weight_loader)
    weight_loader(param, self.model.embed_tokens.weight)
```

This can probably be copy-pasted to other LM classes other than `LlamaForCausalLM`, but I'm less familiar with those, and will defer to maintainers to decide whether to do the copying.

The effect can be tested by running `python3 -m sglang.bench_latency --correct --model HuggingFaceTB/SmolLM-135M-Instruct`. This would output gibberish before the change, but output coherent answers after it.

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [x] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: python/sglang/srt/models/llama.py [hunk 1]
```diff
@@ -403,6 +403,12 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
                 weight_loader = getattr(param, "weight_loader", default_weight_loader)
                 weight_loader(param, loaded_weight)
 
+        if hasattr(self.config, "tie_word_embeddings") and self.config.tie_word_embeddings:
+            # Tie output embedding layer to input embedding layer, to solve issues where lm_head.weight is missing
+            param = self.lm_head.weight
+            weight_loader = getattr(param, "weight_loader", default_weight_loader)
+            weight_loader(param, self.model.embed_tokens.weight)
+
```
Comment (merrymercy) [inline python/sglang/srt/models/llama.py]:
```suggestion
        if (
            hasattr(self.config, "tie_word_embeddings")
            and self.config.tie_word_embeddings
        ):
            # Tie output embedding layer to input embedding layer, to solve issues where lm_head.weight is missing
            param = self.lm_head.weight
            weight_loader = getattr(param, "weight_loader", default_weight_loader)
            weight_loader(param, self.model.embed_tokens.weight)
```

File: python/sglang/srt/models/llama.py [hunk 2]
```diff
@@ -403,6 +403,14 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
...
                 weight_loader(param, loaded_weight)
 
+        if (
+            hasattr(self.config, "tie_word_embeddings")
+            and self.config.tie_word_embeddings
+        ):
+            # Tie output embedding layer to input embedding layer, to solve issues where lm_head.weight is missing
+            param = self.lm_head.weight
+            weight_loader = getattr(param, "weight_loader", default_weight_loader)
+            weight_loader(param, self.model.embed_tokens.weight)
```
Comment (merrymercy) [inline python/sglang/srt/models/llama.py:413]:
```suggestion
            weight_loader(param, self.model.embed_tokens.weight)
            
```

File: python/sglang/srt/models/llama.py [hunk 3]
```diff
@@ -403,6 +403,15 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
...
+
```
Comment (merrymercy) [inline python/sglang/srt/models/llama.py]:
```suggestion

```

**Comment by TianyiQ** [comment]
Oops sorry about the failed test; I thought that was due to my local env (it's a bit messed up atm), but looks like it's not.

**Comment by merrymercy** [comment]
@TianyiQ Thanks. It is merged and it passes all tests

**Pull request merged**